apps/aecore/src/aec_peers.erl

```diff
@@ -1272,10 +1315,12 @@ on_connect_timeout(_Ref, State) ->
     epoch_sync:info("Connect timeout with invalid reference", []),
     State.
 
--spec on_tcp_probe_timeout(reference(), state()) -> state().
-on_tcp_probe_timeout(Ref, #state{ tcp_probe_ref = Ref } = State) ->
-    tcp_probe(State#state{ tcp_probe_ref = undefined });
-on_tcp_probe_timeout(_Ref, State) ->
+-spec on_tcp_probe_timeout(reference(), peer_pool_name(), state()) -> state().
+on_tcp_probe_timeout(Ref, verified, #state{ tcp_verified_probe_ref = Ref } = State) ->
+    tcp_probe(verified, State#state{ tcp_verified_probe_ref = undefined });
+on_tcp_probe_timeout(Ref, unverified, #state{ tcp_unverified_probe_ref = Ref } = State) ->
+    tcp_probe(unverified, State#state{ tcp_unverified_probe_ref = undefined });
+on_tcp_probe_timeout(_Ref, _, State) ->
     epoch_sync:info("TCP probe timeout with invalid reference", []),
```

Comment:
Perhaps also include the `PeerPoolName` here as well, since now the `Ref` could be valid, but the pool name wrong...